cmd/diff.go

```diff
@@ -108,27 +98,34 @@ func diffImages(image1Arg, image2Arg string, diffArgs []string) error {
 
 	diffTypes, err := differs.GetAnalyzers(diffArgs)
 	if err != nil {
-		glog.Error(err.Error())
-		return errors.New("Could not perform image diff")
+		return fmt.Errorf("Could not perform image diff: %s", err.Error())
 	}
 
 	req := differs.DiffRequest{*imageMap[image1Arg], *imageMap[image2Arg], diffTypes}
 	diffs, err := req.GetDiff()
 	if err != nil {
-		glog.Error(err.Error())
-		return errors.New("Could not perform image diff")
+		return fmt.Errorf("Could not perform image diff: %s", err.Error())
 	}
 	glog.Info("Retrieving diffs")
 	outputResults(diffs)
 
 	if save {
 		glog.Infof("Images were saved at %s and %s", imageMap[image1Arg].FSPath,
 			imageMap[image2Arg].FSPath)
-
 	}
 	return nil
 }
 
+func processImageName(imageName string) string {
```

Comment:
i would get rid of the nested negated conditions

```golang
  if pkgutil.IsTar(imageName) || strings.HasPrefix(imageName, "daemon://") return imageName
  // default condition
  return "remote://" + imageName
```

or even

```golang
if pkgutil.IsTar(imageName)
 return ...
if strings.HasPrefix(imageName, "daemon://")
 return ...

return ...
```

Reply:
updated